ntbs-service/Helpers/SessionStateHelper.cs

```diff
@@ -12,8 +12,8 @@ public static void UpdateSessionActivity(ISession session)
 
         public static bool IsUpdatedRecently(ISession session)
         {
-            var date = DateTime.Parse(session.GetString("LastActivityTimestamp"));
-            return date.AddMinutes(30) > DateTime.Now;
+            var activityTimestamp = session.GetString("LastActivityTimestamp");
+            return activityTimestamp == null || DateTime.Parse(activityTimestamp).AddMinutes(30) > DateTime.Now;
```

Comment:
Do you know how we can end up in a state where `LastActivityTimestamp` isn't set? It looks like every request (except for heartbeat requests) should be updating this.

It feels as though the default here ought to be `false` rather than `true` but I do think we need to understand the question above to decide the proper course of action.

Reply:
As I said on the call in startup our middleware was not being set before our endpoint setup and therefore not getting invoked properly. So this issue has been fixed, but I have kept in the check for null (and defaulted to false).